### PR TITLE
build: migrate to owlbot

### PR DESCRIPTION
**Pull request by Neenu1995**

### CODE REVIEW

File: google-cloud-spanner/clirr-ignored-differences.xml [hunk 1]
```diff
@@ -1,662 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (thiagotnunes) [inline google-cloud-spanner/clirr-ignored-differences.xml:1]: Don't we need this anymore?

**Review by thiagotnunes — COMMENTED** [review]
Since we removed the `synth.py` now, how do we regenerate proto changes manually? We need this in order to test features before they are released.

What we used to do is the following:

1. git clone https://github.com/googleapis/googleapis.git
2. Make proto changes in googleapis
3. export SYNTHTOOL_GOOGLEAPIS=<path to googleapis>
4. go to java-spanner repository and run python3 -m synthtool

File: google-cloud-spanner/clirr-ignored-differences.xml (hunk 1 above)
Comment (Neenu1995) [reply google-cloud-spanner/clirr-ignored-differences.xml:1]: Once the version with the breaking changes is published, the clirr check won't detect the same changes as breaking again. So the clirr-ignored-differences file is no longer required to ignore the same breaking changes.

File: samples/snippets/src/main/java/com/example/spanner/AsyncQueryExample.java [hunk 2]
```diff
@@ -71,18 +71,18 @@ public CallbackResponse cursorReady(AsyncResultSet resultSet) {
                   try {
                     while (true) {
                       switch (resultSet.tryNext()) {
-                        // OK: There is a row ready.
+                          // OK: There is a row ready.
```
Comment (thiagotnunes) [inline samples/snippets/src/main/java/com/example/spanner/AsyncQueryExample.java:74]: @Neenu1995 it seems like owlbot is wrongly formatting these comments (whenever it re-runs). We don't have any special checkstyle configuration, so even though we can manually fix this, owlbot will break it whenever it re-runs.

Is there anything we can do in owlbot to fix this?

File: google-cloud-spanner/clirr-ignored-differences.xml (hunk 1 above)
Comment (thiagotnunes) [reply google-cloud-spanner/clirr-ignored-differences.xml:1]: I see, but it is nice to have this file with some examples if we have to add any exceptions. Could we leave a few entries here? Or is this automatically deleted?

**Review by lesv — APPROVED** [review]

**Comment by Neenu1995** [comment]
Closing this in favour of #1569

**Pull request closed without merging**